src/TritonRoute/src/dr/FlexDR_end.cpp

```diff
@@ -368,7 +368,7 @@ void FlexDRWorker::endAddNets_merge(frDesign* design,
         }
         frPoint bp, ep;
         ps->getPoints(bp, ep);
-        if (bp == pt || ep == pt) {
+        if (ps->intersectsCenterLine(pt)) {
```

Comment:
When does it happen that you are on the center line but not the end point?

Reply:
TR has a very loose (non-standard) routing model so this is allowable.  In most tools you would have to split the segment and insert a steiner point in this case.